### DISCUSSION TITLE
updateMany is only executed when awaited or chained with .then - Intended?

### DISCUSSION THREAD

**Discussion by leumasme** [prisma/prisma#7033] [Q&A] [answered]
It seems that my prisma updateMany statement is only actually executed if i either await it or chain it with a .then(...)
When not doing either of these things, it simply won't make any changes to the database.
My process keeps running from a different library even if i dont await it, so that isn't the cause.
Is this intended behavior?
If yes, why?

**Comment by matthewmueller** [comment, marked as answer]
It's intended. It's to support queries like the following in the [fluent API](https://www.prisma.io/docs/concepts/components/prisma-client/relation-queries#fluent-api):

```js
const postsByUser: Post[] = await prisma.user
  .findUnique({ where: { email: 'alice@prisma.io' } })
  .posts()
```

In JS, these are called [thenables](https://masteringjs.io/tutorials/fundamentals/thenable). Definitely confusing if you forget the await, but there's not any other way to do chaining like that.
